packages/eslint-plugin-choreography-ts/src/no-unused-colocally-location.ts

```diff
@@ -0,0 +1,166 @@
+                    node: arg,
+                    messageId: "warning",
+                    data: {
+                      location: arg.value,
+                    },
+                  },
+                });
+              }
+            });
+          }
+        }
+      },
+      // Check whether any location is used an argument to a nested `colocally` expression
+      // and also add the locations to `Locations` for checking their use inside the nested `colocally` body
+      [colocallyColocallyUseLocater]: function (node: TSESTree.CallExpression) {
+        const args = node.arguments;
+        if (args[0]?.type === AST_NODE_TYPES.ArrayExpression) {
+          args[0].elements.forEach((location) => {
+            if (location?.type === AST_NODE_TYPES.Literal) {
+              Locations = Locations.filter(
+                (value) => value.location !== location.value
+              ); // Remove locations seen in the nested `colocally` call from a potential parent `colocally` call
+              Locations.push({
+                location: location.value as string,
+                report: {
+                  node: location,
+                  messageId: "warning",
+                  data: {
+                    location: location.value,
+                  },
+                },
+              });
+            }
+          });
+        }
+      },
+      // Check whether any location is used as an argument to a nested `locally expression
+      [colocallyLocallyUseLocater]: function (node: TSESTree.CallExpression) {
+        const args = node.arguments;
+        if (args[0]?.type === AST_NODE_TYPES.Literal) {
+          Locations = Locations.filter(
+            (value) => value.location !== (args[0] as TSESTree.Literal).value
+          ); // Remove locations seen in nested `locally` call
+        }
+      },
+      // Check whether any location is used as an argument to a nested `multicast` expression
+      [colocallyMulticastUseLocater]: function (node: TSESTree.CallExpression) {
+        const args = node.arguments;
+        // Check whether any location is the sender
+        if (args[0]?.type === AST_NODE_TYPES.Literal) {
+          Locations = Locations.filter(
+            (value) => value.location !== (args[0] as TSESTree.Literal).value
+          ); // Remove locations used as the sender
+        }
+        // Check whether any location is a receiver
+        if (args[1]?.type == AST_NODE_TYPES.ArrayExpression) {
+          args[1].elements.forEach((receiver) => {
+            if (receiver?.type === AST_NODE_TYPES.Literal) {
+              Locations = Locations.filter(
+                (value) => value.location !== receiver.value
+              ); // Remove locations used as a receiver
+            }
+          });
+        }
+      },
+      // Check whether any `broadcast` is used inside the `colocally` call, which instantly designates the locations as being used
+      [colocallyBroadcastUseLocater]: function () {
```

Comment:
We might have discussed this, but now I believe we also need to consider the `call` operator and mark all locations as used for that case. 

```ts
import { Choreography, Located } from "@choreography-ts/core";

const subChoreo: Choreography<
  "alice" | "bob" | "carol",
  [],
  [Located<string, "bob">]
> = async ({ locally }) => {
  const msgAtBob = await locally("bob", () => "Hi from bob");
  return [msgAtBob];
};

type Locations = "alice" | "bob" | "carol";
const test: Choreography<Locations, [], []> = async ({ colocally }) => {
  await colocally(
    ["alice", "bob", "carol"],
    async ({ comm, call }) => {
      const [msgAtBob] = await call(subChoreo, []);
      return [];
    },
    []
  );
  return [];
};

```

In the above code, `subChoreo` mentions all three locations, but because it is not "visible" from inside the choreography, it reports that all three locations are not used, which is not the case.

Reply:
Ok, this is done now.